flux/src/main/java/software/amazon/aws/clients/swf/flux/util/ThreadUtils.java

```diff
@@ -31,14 +32,41 @@
 
     private ThreadUtils() {}
 
+    /**
+     * This is a copy of java.util.concurrent.Executors.DefaultThreadFactory, but with the pool name overridden.
+     */
+    static class NamedThreadFactory implements ThreadFactory {
```

Comment:
It would require another dependency to be added (like Guava or something), but would it be better to use a version of this class maintained by someone else? I can't imagine that this implementation would really change, but I suppose it could.

Reply:
It's worth looking into in the future; for now I'm trying to avoid adding dependencies where it's reasonable to do so.